add_to_app/flutter_module_books/lib/main.dart

```diff
@@ -0,0 +1,191 @@
+import 'package:flutter/material.dart';
+// This file is autogenerated from the schema file at pigeon/schema.dart.
+// Regenerate by running (from flutter_module_books):
+// flutter pub run pigeon \
+//     --input pigeon/schema.dart \
+//     --java_out ../android_books/app/src/main/java/dev/flutter/example/books/Api.java \
+//     --java_package "dev.flutter.example.books"
+import 'package:flutter_module_books/api.dart';
+
+void main() => runApp(MyApp());
+
+class MyApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      theme: ThemeData(
+        primaryColor: const Color(0xff6200ee),
+      ),
+      home: BookDetail(),
+    );
+  }
+}
+
+typedef BookReceived = void Function(Book book);
+
+class FlutterBookApiHandler extends FlutterBookApi {
+  FlutterBookApiHandler(this.callback);
+
+  final BookReceived callback;
+
+  @override
+  void displayBookDetails(Book book) {
+    assert(
+        book != null,
+        'Non-null book expected from FlutterBookApi.displayBookDetails call.',
+    );
+    callback(book);
+  }
+}
+
+class BookDetail extends StatefulWidget {
+  @override
+  _BookDetailState createState() => _BookDetailState();
+}
+
+class _BookDetailState extends State<BookDetail> {
+  Book book;
+
+  HostBookApi hostApi;
+
+  FocusNode textFocusNode = FocusNode();
+  TextEditingController titleTextController = TextEditingController();
+  TextEditingController subtitleTextController = TextEditingController();
+  TextEditingController authorTextController = TextEditingController();
+
+  @override
+  void initState() {
+    super.initState();
+
+    // This `HostBookApi` class instance lets us make outgoing calls to the platform.
+    hostApi = HostBookApi();
+
+    // Registering this `FlutterBookApiHandler` class lets us receive incoming
+    // calls from the platform.
+    FlutterBookApi.setup(FlutterBookApiHandler(
+      // The `FlutterBookApi` just has one method. Just give a closure for that
+      // method to the handler class.
+      (Book book) {
+        setState(() {
+          // This book model is what we're going to return to Kotlin eventually.
+          // Keep it bound to the UI.
+          this.book = book;
+          titleTextController.text = book.title;
+          titleTextController.addListener(() {
+            this.book?.title = titleTextController.text;
+          });
+          subtitleTextController.text = book.subtitle ?? '';
```

Comment:
The `??`operator sticks out a bit here. Is it possible to correct for this (substitute an empty string for null) when the book record is first downloaded?

Reply:
ya, I thought about it a bit when writing too. Though I thing generally, expecting the contract to be in the other side's implementation isn't right. The contract should be in the schema. Though in this case, it means https://github.com/flutter/flutter/issues/59118 :D

Reply:
Yeah, this becomes a different thing entirely once NNBD lands.

Reply:
ya, hopefully it's a good thing. Since we don't want to create custom annotations for Kotlin and Swift currently to make those sides more ergonomic. We can then piggy back off of the dart NNBD syntax once it matures a bit.